R/names.R

```diff
@@ -193,15 +193,47 @@ detect_dot_dot <- function(names) {
   grep("^[.][.](?:[.]|[1-9][0-9]*)$", names)
 }
 
-#' Extract repaired names from a vector
+#' Get or set the names of a vector
 #'
-#' Returns the repaired names from a vector, even if the vector is unnamed.
+#' @description
+#' These functions work like [rlang::names2()], [names()] and [names<-()],
+#' except that they return or modify the conceptual names of the vector.
```

Comment:
the rowwise names of the vector. These are:

* The usual `names()` for atomic vectors and lists
* The row names for data frames and matrices
* The names of the first dimension for arrays

Reply:
Excuse the drive-by comment, but what is meant by "conceptual" names? If that term is defined and used elsewhere and I just don't have enough context, then never mind.

Reply:
Agreed this is not a defined term. I'm suggesting to replace "conceptual" by "rowwise", which seems clearer?

Reply:
I think since it is so related to the vctrs concept of size, they could be called "size consistent names"